### PR TITLE
Pre-commit formatting for Python binding

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,9 +9,10 @@ repos:
   - id: check-toml
   - id: check-yaml
   - id: check-json
-    exclude: .evergreen/atlas_data_lake/test.driverdata.json
+    exclude: '.evergreen/atlas_data_lake/test.driverdata.json'
```
Comment (blink1073) [inline .pre-commit-config.yaml:12]: Let's revert changes to this file and `pyproject.toml` and make them part of the PR that adds `.git-ignore-revs`.

Comment (NoahStapp) [reply .pre-commit-config.yaml:12]: The `pyproject.toml` changes are adding the ruff rules, wouldn't we want that in this PR?

Comment (blink1073) [reply .pre-commit-config.yaml:12]: So that it doesn't get ignored as part of this commit.

Comment (NoahStapp) [reply .pre-commit-config.yaml:12]: Ah right!

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**